File: backend/app/schemas/operador.py
from datetime import date, datetime
from pydantic import BaseModel, ConfigDict


class OperadorBase(BaseModel):
    id_usuario: int
    alias: str
    numero_licencia: str | None = None
    licencia_vigencia: date | None = None
    sua: str | None = None
    sua_vigencia: date | None = None
    estudio_medico: date | None = None
    activo: bool = True


class OperadorCreate(OperadorBase):
    pass


class OperadorUpdate(BaseModel):
    id_usuario: int | None = None
    alias: str | None = None
    numero_licencia: str | None = None
    licencia_vigencia: date | None = None
    sua: str | None = None
    sua_vigencia: date | None = None
    estudio_medico: date | None = None
    activo: bool | None = None


class OperadorResponse(OperadorBase):
    id_operador: int
    created_at: datetime
    updated_at: datetime

    model_config = ConfigDict(from_attributes=True)

File: frontend/src/types/operador.ts
export type Operador = {
  id_operador: number;
  id_usuario: number;
  alias: string;
  numero_licencia: string | null;
  rfc: string | null;
  curp: string | null;
  numero_expediente_medico: string | null;
  licencia_vigencia: string | null;
  sua: string | null;
  sua_vigencia: string | null;
  estudio_medico: string | null;
  activo: boolean;
  created_at: string;
  updated_at: string;
};

export type CreateOperadorPayload = {
  id_usuario: number;
  alias: string;
  numero_licencia?: string | null;
  rfc?: string | null;
  curp?: string | null;
  numero_expediente_medico?: string | null;
  licencia_vigencia?: string | null;
  sua?: string | null;
  sua_vigencia?: string | null;
  estudio_medico?: string | null;
  activo: boolean;
};

export type UpdateOperadorPayload = {
  id_usuario?: number | null;
  alias?: string | null;
  numero_licencia?: string | null;
  rfc?: string | null;
  curp?: string | null;
  numero_expediente_medico?: string | null;
  licencia_vigencia?: string | null;
  sua?: string | null;
  sua_vigencia?: string | null;
  estudio_medico?: string | null;
  activo?: boolean;
};
